cmd/frontend/graphqlbackend/external_services.go

```
@@ -159,6 +161,129 @@ func (r *schemaResolver) UpdateExternalService(ctx context.Context, args *update
+	if err != nil {
+		return "", err
+	}
+
+	switch c := config.(type) {
+	case *schema.AWSCodeCommitConnection:
+		exclusion := &schema.ExcludedAWSCodeCommitRepo{Id: strconv.FormatInt(int64(repository.ID), 10), Name: string(repository.Name)}
+		if !schemaContainsExclusion(c.Exclude, exclusion) {
+			c.Exclude = append(c.Exclude, &schema.ExcludedAWSCodeCommitRepo{Id: strconv.FormatInt(int64(repository.ID), 10), Name: string(repository.Name)})
+		}
+	case *schema.BitbucketCloudConnection:
+		exclusion := &schema.ExcludedBitbucketCloudRepo{Name: string(repository.Name)}
+		if !schemaContainsExclusion(c.Exclude, exclusion) {
+			c.Exclude = append(c.Exclude, &schema.ExcludedBitbucketCloudRepo{Name: string(repository.Name)})
+		}
+	case *schema.BitbucketServerConnection:
+		exclusion := &schema.ExcludedBitbucketServerRepo{Id: int(repository.ID), Name: string(repository.Name)}
+		if !schemaContainsExclusion(c.Exclude, exclusion) {
+			c.Exclude = append(c.Exclude, &schema.ExcludedBitbucketServerRepo{Id: int(repository.ID), Name: string(repository.Name)})
+		}
+	case *schema.GitHubConnection:
+		exclusion := &schema.ExcludedGitHubRepo{Id: strconv.FormatInt(int64(repository.ID), 10), Name: string(repository.Name)}
+		if !schemaContainsExclusion(c.Exclude, exclusion) {
+			c.Exclude = append(c.Exclude, &schema.ExcludedGitHubRepo{Id: strconv.FormatInt(int64(repository.ID), 10), Name: string(repository.Name)})
+		}
+	case *schema.GitLabConnection:
+		exclusion := &schema.ExcludedGitLabProject{Name: string(repository.Name)}
+		if !schemaContainsExclusion(c.Exclude, exclusion) {
+			c.Exclude = append(c.Exclude, &schema.ExcludedGitLabProject{Name: string(repository.Name)})
+		}
+	case *schema.GitoliteConnection:
+		exclusion := &schema.ExcludedGitoliteRepo{Name: string(repository.Name)}
+		if !schemaContainsExclusion(c.Exclude, exclusion) {
+			c.Exclude = append(c.Exclude, &schema.ExcludedGitoliteRepo{Name: string(repository.Name)})
+		}
+	default:
+		return "", err
```

Isn't `err` going to be `nil` here? 🤔
We should return a custom error maybe?

---

in fact we shouldn't return an error here, there should be no default, we just need to return the same config